armada_jupyter/submissions.py

```diff
@@ -0,0 +1,205 @@
+from typing import List, Optional, Dict, Union
+
+import yaml
+
+from armada_jupyter.constants import (
+    YMLSTR,
+    DEFAULT_QUEUE,
+    DEFAULT_PRIORITY,
+    DEFAULT_TIMEOUT,
+    DEFAULT_MEMORY,
+    DEFAULT_CPU,
+)
+
+
+class K8sResourceOptions:
+    """
+    Manages the options for a kubernetes resource request or limit
+    """
+
+    def __init__(
+        self,
+        cpu: int,
+        memory: str,
+        nvidia_gpu: Union[str, int, None] = None,
```

Comment:
I don't really know if supporting both at same time is necessary.  One or the other.

Reply:
I will change this :))